File: webapi/Models/math/Coefficient.cs
namespace webapi.Models.Math
{
    public class Coefficient
    {
        public dynamic Value { get; set; }

        public NumberType Type { get; set; }

        public Coefficient() { }

        public Coefficient(int value)
        {
            Value = value;
            Type = NumberType.Number;
        }

        public Coefficient(double value)
        {
            Value = value;
            Type = NumberType.Number;
        }

        public Coefficient(string value)
        {
            Value = value;
            Type = NumberType.Fraction;
        }
    }

    public enum NumberType
    {
        Number,
        Fraction
    }
}

File: reactapp/src/types/math/coefficient.ts
import { NumberType } from "./enums.ts";

export interface ICoefficient {
    value: number | string;
    type: NumberType;
}

export class Coefficient implements ICoefficient {
    constructor(
        public value: number | string,
        public type: NumberType
    ) {}

    public static createNumberCoefficient(value: number): Coefficient {
        return new Coefficient(value, NumberType.Number);
    }
}
